ground/src/main/java/com/google/android/ground/ui/map/gms/GoogleMapsFragment.kt

```diff
@@ -96,14 +94,16 @@ class GoogleMapsFragment : SupportMapFragment(), MapFragment {
    * References to Google Maps SDK Markers present on the map. Used to sync and update polylines
    * with current view and data state.
    */
-  private val markers: MutableMap<Marker, MapLocationOfInterest> = HashMap()
+  private val clusters: MutableMap<LocationOfInterestClusterItem, MapLocationOfInterest> = HashMap()
```

Comment:
Does the MapAdapter need to know about LOIs? Or should it operate directly on MapPin and other map features?

Reply:
I can try renaming this, but the "ClusterItem" is an inheritor of a type the cluster manger needs to operate on. Instead of taking LOIs, I can have it operate on MapLocationOfInterest

Reply:
Sounds like I may have opened up a can of worms. Let's come back to this in a follow-up!